packages/client/src/components/interface/Header/Header.tsx

```diff
@@ -288,11 +307,38 @@ class HeaderComp extends React.Component<IProps, IState> {
     this.props.redirectToAuthentication()
   }
 
+  isLandingPage = () => {
+    const role = this.props.userDetails && this.props.userDetails.role
+    const location = this.props.history.location.pathname as any
+    if (
+      FIELD_AGENT_ROLES.includes(role as string) &&
+      [HOME].includes(location)
+    ) {
+      return true
+    } else if (
+      NATL_ADMIN_ROLES.includes(role as string) &&
+      [PERFORMANCE_HOME].includes(location)
+    ) {
+      return true
+    } else if (
+      SYS_ADMIN_ROLES.includes(role as string) &&
+      [OPERATIONAL_REPORT].includes(location)
+    ) {
+      return true
+    } else if (
+      REGISTRAR_ROLES.includes(role as string) &&
+      [REGISTRAR_HOME].includes(location)
```

Comment:
I think you have to extract this to a function and use that for other similar checks

Reply:
I have returned a single true or false boolean from `isLandingPage` function.